src/core/unit.c

```diff
@@ -3813,7 +3813,7 @@ bool unit_need_daemon_reload(Unit *u) {
         if (fragment_mtime_newer(u->source_path, u->source_mtime, false))
                 return true;
 
-        if (u->load_state == UNIT_LOADED)
+        if (UNIT_IS_LOAD_COMPLETE(u->load_state))
```

Comment:
Well, this was changed intentionally in 19a44dfe4525ab01caf593a9c2beada4b412910d, since drop-ins really doesn't matter for masked units. Instead, let's skip the check for drop-ins entirely (I have a commit ready in https://github.com/YHNdnzj/systemd/commit/d7474c816032a4d6690e3112a8d0fe9da3136ab8, but didn't submit it. Consider just cherry-picking that?)

Reply:
Done.